src/libsyntax/parse/token.rs

```diff
@@ -508,14 +502,8 @@ impl Token {
         }
     }
 
-    pub fn interpolated_to_tokenstream(&self, sess: &ParseSess, span: Span)
-        -> TokenStream
-    {
-        let nt = match *self {
-            Token::Interpolated(ref nt) => nt,
-            _ => panic!("only works on interpolated tokens"),
-        };
-
+    pub fn interpolated_to_tokenstream(sess: &ParseSess, nt: Lrc<Nonterminal>, span: Span)
+                                       -> TokenStream {
```

Comment:
Looks like this can be made a method on `Nonterminal` now.

Reply:
I tried that but it didn't work because `nt` is `Lrc<Nonterminal>` rather than `Nonterminal` -- this func can take ownership of the `Rc` but not the `Nonterminal` itself.

Reply:
Hmm, everything inside it seems to work by ref, why does it need to take ownership?

Reply:
Hmm, maybe I tried it before I got rid of the `nt.1.force()` code. I'll try again on Monday.

Reply:
Confirmed: I tried to make the change as part of the first commit, which doesn't work. But it does work as an extra commit at the end.